pkg/envoy/ciliumenvoyconfig.go

```diff
@@ -145,8 +145,14 @@ func ParseResources(cecNamespace string, cecName string, anySlice []cilium_v2.XD
 				listener.EnableReusePort = &wrapperspb.BoolValue{Value: false}
 			}
 
+			// Figure out if this is an internal listener
+			isInternalListener := listener.GetInternalListener() != nil
+
+			// Only inject Cilium filters if Cilium allocates listener address
+			injectCiliumFilters := listener.GetAddress() == nil && !isInternalListener
+
 			// Inject Cilium bpf metadata listener filter, if not already present.
-			{
+			if !isInternalListener {
```

Comment:
Would there be any reason to validate the parsed resource to ensure that any listeners are not configured with the bpf_metadata filter?

Reply:
We plan to add metadata passing over internal listeners at a later point, but do not know yet if that will involve use of the `bpf_metadata` filter in the internal listener config. Hence I'd think clamping this down at this point is premature.